Doc/library/subprocess.rst

```diff
@@ -356,6 +356,15 @@ functions.
    arguments for additional differences from the default behavior.  Unless
    otherwise stated, it is recommended to pass *args* as a sequence.
 
+   .. warning::
+
+      Using the string ``"python"`` as the program to execute is not a good
```

Comment:
I personally feel like we should not single out `python` executable because wouldn't the same apply with any other program? IMHO we should point out that on Windows non-absolute path executables don't resolve by using the `PATH` environment variable and as such we encourage users to pass executables as absolute paths. And then we can mention to use `shutil.which` if they want a discover executable using the `PATH` environment variable, and in particular if they want to execute a script with the running python `sys.executable`. Wouldn't that be more accurate and general?

Reply:
It will only apply to anything inside the same directory as python.exe, which is basically just `python`. In a venv, it's also anything that installed a script wrapper, so `pip` or `virtualenv`, among others.

That said, it's been "broken" for 2-3 full releases now (depending on how you count), and the only issues I've heard of were people who thought "python" would be found on PATH. So I don't think switching it back is super important. Arguably, we made it closer to "search PATH".